src/wp-includes/class-wp-scripts.php

```diff
@@ -907,9 +907,19 @@ private function is_delayed_strategy( $strategy ) {
 	 * @return string The best eligible loading strategy.
 	 */
 	private function get_eligible_loading_strategy( $handle ) {
-		$eligible = $this->filter_eligible_strategies( $handle );
+		$intended = (string) $this->get_data( $handle, 'strategy' );
 
-		// Bail early once we know the eligible strategy is blocking.
+		// Bail early if there is no intended strategy.
+		if ( ! $intended ) {
+			return '';
+		}
+
+		// If the intended strategy is 'defer', limit the initial list of eligibles.
```

Comment:
```suggestion
		// If the intended strategy is 'defer', limit the initial list of eligibles since 'async' can fallback to 'defer' but not vice-versa.
```

Reply:
Improved in https://github.com/WordPress/wordpress-develop/pull/4852/commits/edd93474667b09d03ec899c840a687b5cd5192aa.